firmware/application/apps/ui_text_editor.cpp

```diff
@@ -221,28 +222,45 @@ void TextViewer::paint_text(Painter& painter, uint32_t line, uint16_t col) {
                  r.top() + (int)i * char_height,
                  clear_width * char_width, char_height},
                 style().background);
+
+        // if cursor line got overwritten, disable XOR of old cursor when displaying new cursor
+        if (i == paint_state_.line)
```

Comment:
If the cursor position was inside a line printed on the screen in the normal font , it would get XOR'd when the cursor is moved to a new position, causing the old character position to be highlighted.  Result was that old cursor positions were still highlighted in some cases.  My workaround was to forget the old cursor position.